### PR TITLE
.gitignore + make in-tree builds work

### PR DESCRIPTION
**Pull request by kainino0x**
This somewhat reduces noise for in-tree builds, too, in case any contributor wants to or accidentally does that.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,119 @@
+generator/
```
Comment (Kangz) [inline .gitignore]: This should be more specific as this will ignore files added to templates etc.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,10 @@
-generator/
+generator/mock
```
Comment (Kangz) [inline .gitignore]: Can we change the directory where things are generated instead to be like "generator/generated" ?

**Comment by kainino0x** [comment]
I was trying to test the .gitignore for in-tree builds and realized the in-tree build actually doesn't work on mac/linux because of a filename conflict. Pushed a fix.

However this actually causes existing partial builds to break if you don't remove build/examples/glTFViewer first (because the new output file is build/examples/glTFViewer/glTFViewer). What's your preference on allowing in-tree builds? I could just omit that change.

**Comment by kainino0x** [comment]
np, can you mark as approved?

**Pull request merged**